### PR TITLE
HPCC-12993 Fix gcc warnings

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
See also https://github.com/hpcc-systems/LN/pull/727

**Comment by ghalliday** [comment]
Interestingly a warning from the lexer that FORMAT was never matched revealed a regression in an (unusual) option for FORMAT on a CSV HEADER.  I'll comment any unusual changes.

File: ecl/hql/hqllex.l [hunk 1]
```diff
@@ -705,10 +705,9 @@ FILTERED            { RETURNSYM(FILTERED); }
 FIRST               { RETURNSYM(FIRST); }
 FIXED               { RETURNSYM(TOK_FIXED); }
 FLAT                { RETURNSYM(FLAT); }
-FORMAT              { RETURNSYM(FORMAT); }
-FROM                { RETURNSYM(FROM); }
```
Comment (ghalliday) [inline ecl/hql/hqllex.l]: Wasn't alphabetical, so was easy for someone to miss that it was already present.

File: ecl/hqlcpp/hqlcse.cpp [hunk 2]
```diff
@@ -106,6 +106,7 @@ bool canCreateTemporary(IHqlExpression * expr)
     case no_thisnode:
     case no_libraryscopeinstance:
     case no_loopbody:
+    case no_external:
```
Comment (ghalliday) [inline ecl/hqlcpp/hqlcse.cpp]: This fixes csvheader4.ecl which was broken, but I hadn't picked up on the regression suite differences.  Possibly a result of taking too many changes at once.

File: system/jlib/jerror.hpp [hunk 3]
```diff
@@ -32,6 +32,7 @@
 #define JLIBERR_BadUtf8InArguments              6001
 #define JLIBERR_InternalError                   6002
 #define JLIBERR_CppCompileError                 6003
+#define JLIBERR_FailToImpersonate               6004
```
Comment (richardkchapman) [inline system/jlib/jerror.hpp]: Did you intend to use this in jutil.cpp?

**Comment by richardkchapman** [comment]
I don't think it is wise to mix the changes in the lexer that fix an obscure bug with the changes elsewhere that merely remove warnings. Should be split into two PRs/commits

File: system/jlib/jerror.hpp (hunk 3 above)
Comment (ghalliday) [inline system/jlib/jerror.hpp]: I did, but I guess then used the Os exception

**Comment by ghalliday** [comment]
I'll clean it up when I get back

**Comment by ghalliday** [comment]
@richardkchapman now contains just warning fixes

**Pull request merged**